app/models/project.rb

```diff
@@ -322,6 +318,17 @@ def self.visible_by(user = nil)
     end
   end
 
+  # Returns a ActiveRecord::Relation to find all projects for which +user+ has the given +permission+
+  #
+  # Valid options:
+  # * :project => limit the condition to project
+  # * :with_subprojects => limit the condition to project and its subprojects
+  # * :member => limit the condition to the user projects
+  def self.allowed_to(user, permission, options = {})
+    where(allowed_to_condition(user, permission, options))
+      .references(:projects)
```

Comment:
@ulferts you're right. it shouldn't be needed - but without it, deprecation warnings are emitted (same problem as https://github.com/opf/openproject/pull/3194#commitcomment-12104671)